bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ThingActionsResource.java

```diff
@@ -230,6 +251,76 @@ public Response executeThingAction(@PathParam("thingUID") @Parameter(description
         }
     }
 
+    private @Nullable ConfigDescriptionParameter convertToConfigDescriptionParameter(Input input) {
+        boolean supported = true;
+        ConfigDescriptionParameter.Type parameterType = ConfigDescriptionParameter.Type.TEXT;
+        String defaultValue = null;
+        boolean required = false;
+        String context = null;
+        switch (input.getType()) {
+            case "boolean":
+                defaultValue = "false";
+                required = true;
+            case "java.lang.Boolean":
+                parameterType = ConfigDescriptionParameter.Type.BOOLEAN;
+                break;
+            case "byte":
+            case "short":
+            case "int":
+            case "long":
+                defaultValue = "0";
+                required = true;
+            case "java.lang.Byte":
+            case "java.lang.Short":
+            case "java.lang.Integer":
+            case "java.lang.Long":
+                parameterType = ConfigDescriptionParameter.Type.INTEGER;
+                break;
+            case "float":
+            case "double":
+                defaultValue = "0";
+                required = true;
+            case "java.lang.Float":
+            case "java.lang.Double":
+                parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                break;
+            case "java.lang.String":
+                parameterType = ConfigDescriptionParameter.Type.TEXT;
+                break;
+            case "java.time.LocalDate":
+                parameterType = ConfigDescriptionParameter.Type.TEXT;
+                context = "date";
```

Comment:
There is AFAIK no date context in the UI, but would probably nice to add one.

Reply:
date is a documented context:
https://github.com/openhab/openhab-core/blob/47284e5521c5bccb0db5afec1202a0aa453a2f4f/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigDescriptionParameter.java#L318

Reply:
Thanks for the link … if it is not in the UI I will add it.
That’s however nothing that matters for this PR, I was just noting that.

Reply:
I am curious to check if this is already used by any binding.

We could also check if this "context" is documented in our documentation and not only in Java doc.

Reply:
time as context value is used by 4 bindings:
```
$ find */src -name "*.xml" -exec grep -H "<context>" {} \; | grep ime
org.openhab.binding.astro/src/main/resources/OH-INF/config/config.xml:                  <context>time</context>
org.openhab.binding.astro/src/main/resources/OH-INF/config/config.xml:                  <context>time</context>
org.openhab.binding.gpio/src/main/resources/OH-INF/thing/pigpio-remote.xml:                             <context>time</context>
org.openhab.binding.gpio/src/main/resources/OH-INF/thing/pigpio-remote.xml:                             <context>time</context>
org.openhab.binding.lametrictime/src/main/resources/OH-INF/thing/device.xml:                            <context>password</context>
org.openhab.binding.robonect/src/main/resources/OH-INF/config/config.xml:                       <context>time</context>
org.openhab.transform.basicprofiles/src/main/resources/OH-INF/config/time-range-command.xml:                    <context>time</context>
org.openhab.transform.basicprofiles/src/main/resources/OH-INF/config/time-range-command.xml:                    <context>time</context>
```

Reply:
And datetime or date is apparently not yet used.

Reply:
So it would be important that you support date and datetime contexts mainly for thing actions.

Reply:
It is already properly documented:
https://www.openhab.org/docs/developer/addons/config-xml.html#supported-contexts

I have to check if formats mentioned in doc match the ISO formats because you will certainly use the documented formats when calling the action. So we have to support such string formats as input.